src/test/java/net/sf/jabref/importer/fileformat/MedlinePlainImporterTest.java

```diff
@@ -0,0 +1,127 @@
+                "MedlinePlainImporterTestEmpty.txt");
+
+        //Test with txt Files which are not in recognized Format
+        for (String str : list) {
+            try (InputStream is = MedlinePlainImporter.class.getResourceAsStream(str)) {
+                Assert.assertFalse(importer.isRecognizedFormat(is));
+            }
+        }
+
+        //Test with txt File which is in recognized Format
+        Assert.assertTrue(importer.isRecognizedFormat(MedlinePlainImporter.class
+                .getResourceAsStream("MedlinePlainImporterTest.txt")));
+
+    }
+
+    /**
+     * test multiple entries in one import file
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testImportEntries() throws Exception {
+
+        //create entries list from the txt file
+        List<BibEntry> entries = importer.importEntries(
+                MedlinePlainImporter.class.getResourceAsStream("MedlinePlainImporterTest.txt"),
+                new OutputPrinterToNull());
+        Assert.assertEquals(7, entries.size());
+
+        //create Bibtexentry from first/second object in the entry list
+        //and test with getField
+        BibEntry test1 = entries.get(0);
+        Assert.assertNull(test1.getField("month"));
+        Assert.assertEquals("Long, Vicky and Marland, Hilary", test1.getField("author"));
+        Assert.assertEquals(
+                "From danger and motherhood to health and beauty: health advice for the factory girl in early twentieth-century Britain.",
+                test1.getField("title"));
+        BibEntry test2 = entries.get(1);
+        Assert.assertEquals("06", test2.getField("month"));
+        Assert.assertNull(test2.getField("author"));
+        Assert.assertNull(test2.getField("title"));
+    }
+
+    @Test
+    public void testEmptyImport() throws Exception {
+        //Test entries with empty txt File
+        List<BibEntry> emptyEntries = importer.importEntries(
+                MedlinePlainImporter.class.getResourceAsStream("MedlinePlainImporterTestEmpty.txt"),
+                new OutputPrinterToNull());
+        Assert.assertEquals(0, emptyEntries.size());
+    }
+
+    /**
+     * test each import entry from a individual file
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testSingleEntries() throws Exception {
+        for (int n = 1; n <= 5; n++) {
+            try (InputStream is = MedlinePlainImporter.class.getResourceAsStream("MedlinePlainImporterTest" + n
+                    + ".txt");
+                    InputStream nis = MedlinePlainImporter.class.getResourceAsStream("MedlinePlainImporterTestBib" + n
+                            + ".bib")) {
+
+                List<BibEntry> entries = importer.importEntries(is, new OutputPrinterToNull());
+                Assert.assertNotNull(entries);
+                Assert.assertEquals(1, entries.size());
+                BibtexEntryAssert.assertEquals(nis, entries.get(0));
+            }
+        }

```

Comment:
You can write it more clearly by separating the input pairs from the logic. Make it obvious what is getting tested. For instance:

```
for (int n = 1; n <= 5; n++) {
    String medlineFile = "MedlinePlainImporterTest" + n + ".txt";
    String bibtexFile = "MedlinePlainImporterTestBib" + n + ".bib";
    assertImportOfMedlineFileEqualsBibtexFile(medlineFile, bibtexFile);
}
```

And the assert method you can implement based on the other code.